src/runtime/CMakeLists.txt

```diff
@@ -185,6 +185,40 @@ if(WITH_LEVEL_ZERO_BACKEND)
         ARCHIVE DESTINATION lib/hipSYCL)
 endif()
 
+if(WITH_METAL_BACKEND)
+  add_library(rt-backend-metal SHARED
+    metal/metal_allocator.cpp
+    metal/metal_backend.cpp
+    metal/metal_hardware_manager.cpp)
+  
+  # TODO: Move this somewhere more appropriate.
+  # TODO: Configure Metal/Metal.hpp in a possibly less intrusive way.
+  # TODO: Consider setting: add_compile_options(-WMTL_IGNORE_WARNINGS)
+  find_library(FOUNDATION_LIB Foundation REQUIRED)
+  find_library(QUARTZ_CORE_LIB QuartzCore REQUIRED)
+  find_library(METAL_LIB Metal REQUIRED)
+  find_library(IOKIT_LIB IOKit REQUIRED)
+  
+  # TODO: Show these dependencies in `syclcc --hipsycl-version`?
+  target_link_libraries(rt-backend-metal PRIVATE
+    ${FOUNDATION_LIB} ${QUARTZ_CORE_LIB} ${METAL_LIB} ${IOKIT_LIB})
+  
+  target_include_directories(rt-backend-metal PRIVATE ${HIPSYCL_SOURCE_DIR}/include)
+  target_link_libraries(rt-backend-metal PRIVATE hipSYCL-rt)
+  
+  target_compile_options(rt-backend-metal PRIVATE ${HIPSYCL_RT_EXTRA_CXX_FLAGS})
+  target_link_libraries(rt-backend-metal PRIVATE ${HIPSYCL_RT_EXTRA_LINKER_FLAGS})
+  
+  if(WITH_SSCP_COMPILER)
+    target_compile_definitions(rt-backend-metal PRIVATE -DHIPSYCL_WITH_SSCP_COMPILER)
+    target_link_libraries(rt-backend-metal PRIVATE llvm-to-spirv)
```

Comment:
Wouldn't you want to link against a future `llvm-to-msl` here (which might internally reuse some SPIR-V infrastructure?)

Reply:
I'm not yet sure how `llvm-to-msl` would work. Hopefully we can share code with `llvm-to-spirv`, then add additional code that structurizes it before `spirv-to-msl`. 

Perhaps the `llvm-to-spirv` code can accept check macros for Level Zero or Metal backends. We might want to enable Apple-specific intrinsics like `simdgroup_matrix_multiply_accumulate` which aren't part of `spirv_ocl`. Conversely, some Level Zero intrinsics might be inappropriate for Apple. We would disable those with Metal but not for ze.

Reply:
Is it appropriate to place the `find_library` invocations in the CMakeLists of `src/runtime`, or do we need to extract it somewhere else? How do/should we mention them in the compile flags listed by `syclcc --hipsycl-version`?

Reply:
Is it appropriate to place the `find_library` invocations in the CMakeLists of `src/runtime`, or do we need to extract it somewhere else? How do/should we mention them in the compile flags listed by `syclcc --hipsycl-version`?